ruby/ql/lib/codeql/ruby/frameworks/Archive.qll

```diff
@@ -0,0 +1,22 @@
+/**
+ * Provides classes for working with archive libraries.
+ */
+
+private import ruby
+private import codeql.ruby.Concepts
+private import codeql.ruby.DataFlow
+private import codeql.ruby.ApiGraphs
+
+/**
+ * Classes and predicates for modelling the RubyZip library
+ */
+module RubyZip {
```

Comment:
@aibaars I think I follow, but is this what you meant by creating a new file for Archive libraries where we can store models for them together?

Reply:
Yes, this looks good. Let's also add `API::getTopLevelMember("Zip").getMember("File").getAnInstantiation()` to cover `Zip::File.new`.